src/commands/update.js

```diff
@@ -0,0 +1,203 @@
+      }
+    }
+  }
+
+  /**
+   * Install the plugins that have  updates, with an interactive install.
+   *
+   * @private
+   * @param {Array<ToUpdatePlugin>} plugins the plugins to update
+   */
+  async __interactiveInstall (plugins) {
+    const inqChoices = plugins
+      .map(plugin => { // map to expected inquirer format
+        return {
+          name: `${plugin.name}@${plugin.latestVersion}`,
+          value: plugin.name
+        }
+      })
+
+    const response = await inquirer.prompt([{
+      name: 'plugins',
+      message: 'Select plugins to update',
+      type: 'checkbox',
+      choices: inqChoices
+    }])
+
+    // install the plugins in sequence
+    for (const plugin of response.plugins) {
+      await this.config.runCommand('plugins:install', [plugin])
+    }
+  }
+
+  /**
+   * Process the plugins, determine if they need updates or warnings.
+   *
+   * @param {object} cliRoot the root folder of the cli
+   * @param {Array<string>} corePlugins package names of the core plugins
+   * @param {Array<InstalledPlugin>} installedPlugins
+   * @returns {Array<ToUpdatePlugin}
+   */
+  async __processPlugins () {
+    const corePlugins = this.config.pjson.oclif.plugins
+    const plugins = []
+
+    // Filter installed plugins:
+    // - remove any plugin that is in core, that is not from the @adobe namespace
+    // These will not be updateable for compatibility reasons
+    const installedPlugins = this.config.plugins.filter(plugin =>
+      !(
+        corePlugins.includes(plugin.name) &&
+        !(plugin.name.startsWith('@adobe/'))
+      )
+    )
+
+    for (const plugin of installedPlugins) {
+      const { type, name, version: currentVersion } = plugin
+      const latestVersion = await getNpmLatestVersion(name)
+      let coreVersion = (type === 'core') ? currentVersion : null
+      const needsUpdate = (latestVersion > currentVersion)
+      let needsWarning = false
+
+      if (!needsUpdate) {
+        // a user installed plugin could be older or the same version as a core plugin it overrides
+        // usually this is not an intended case, and we warn the user
+        // (scenario: a user installs an updated version of the cli with all the updated plugins,
+        // but the user already has these updated as user-installed plugins)
+        const pluginOverridesCorePlugin = (type === 'user' && corePlugins.includes(name))
+        if (pluginOverridesCorePlugin) {
+          coreVersion = await getNpmLocalVersion(this.config.root, name)
+          needsWarning = (coreVersion >= currentVersion)
+        }
+      }
+
+      plugins.push({
+        type,
+        name,
+        currentVersion,
+        latestVersion,
+        coreVersion,
+        needsUpdate,
+        needsWarning
+      })
+    }
+
+    return plugins
+  }
+
+  /**
+   * Command entry point
+   */
+  async run () {
+    const { flags } = this.parse(UpdateCommand)
+    const spinner = ora()
```

Comment:
Why not use `cli-ux.action.start/stop` and not add a dependency on `ora`?

Reply:
actually `cli-ux` is not even a dependency of the base cli (check package.json). It just so happens to be a dependency of a dependency (I'm thinking runtime plugin).  Also, I needed a spinner that "goes away" after use, cli-ux's spinner does not do that.

Reply:
Looks like the `discover` command uses `cli-ux` as well, which is not a dependency as I mentioned. In any case, we need to add `cli-ux` as a dependency here because of `cli.table`.

Reply:
how did it ever work?

Reply:
i mentioned above "dependency of a dependency had cli-ux". so it was in node_modules

Reply:
there is a eslint plugin that would have detected this https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md

Reply:
forgot to mention (See the Slack thread), `aio clear` is now `aio rollback`. I reflected it in the issue description and title when I finalized the PR yesterday.